docs/4.3/enterprise/ssh_rbac.md

```diff
@@ -124,6 +124,11 @@ spec:
       # regular expressions are also supported, for example the equivalent
       # of the list example above can be expressed as:
       'environment': '^test|staging$'
+      # negative matches use the '+not' prefix, this works with both exact
+      # matches and regular expressions.
+      # this is useful when another role may grant extended access and you
+      # don't want to exclude it entirely with a 'deny' rule below.
+      'environment': '+not production'
```

Comment:
The space does feel little strange to me... I'm not 100% sure of the solution but I've written up some examples based on the bug.


### Proposal 1
```
role: +not PROD-*
```
### Proposal 2
```
role: +not(PROD-*)
```
### Proposal 2a
```
role: +not(`PROD-*`)
```
### Proposal 3
```
role: +not_regexp PROD-*
```
### Proposal 4
```
role: +not_regexp(PROD-*)
```
### Proposal 4a
```
role: +not_regexp(`PROD-*`)
```

### Example Group to Roles
```
    - {name: "groups", value: "okta-dev", roles: ["dev"]}
    - {name: "groups", value: "okta-dev", roles: ["+not_regexp(`PROD-*`)"]}
    - {name: "groups", value: "okta-dev", roles: ["+not(PROD-*)"]}
```

```
  'environment': '+not prod*'
```

Reply:
@benarent up to you on the syntax.
Parentheses add complexity to the parsing code. Backticks too, plus I don't see why they are needed.
But if that's beneficial for users, implementation complexity is not an objection.

Reply:
I like `+not_regexp()` over `+not` as it makes it more obvious how the field can be used and what sort of processing is going to apply. The brackets are a cleaner separator than a space IMO.

Reply:
As we've  already `regexp.replace(what, expression, replacement)` https://github.com/gravitational/teleport/issues/3374 

How about? `regexp.not(expression)`

Reply:
I have a few concerns:
- we don't have `regexp.replace` in `node_labels`; `node_labels` are only globs or regexps, without the `{{foo}}`-style substitutions
- `regexp.not(expression)` is a valid regexp on its own; this change would not be strictly backwards-compatible

I'll update the PR to handle `+regexp.not(expr)`.
If you think it's more desirable to stay consistent with other role fields, we can make it `{{regexp.not(expr)}}` and just assume that no customer uses a value that looks like that. Let me know.

Reply:
I think we should stay as consistent as we can, I doubt any customers use a value that looks like this.